### PR TITLE
Fix: Retry CLI launch if needed

### LINKED ISSUE #8212: appengine.standard_python3.bundled-services.mail.wsgi.main_test: test_send_receive failed
This test failed!

To configure my behavior, see [the Flaky Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/main/packages/flakybot).

If I'm commenting on this issue too often, add the `flakybot: quiet` label and
I will stop commenting.

---

commit: 79555e6240e4a72948dab9d06a8587a6cffd7fbb
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/bdafde32-91d5-48a1-8f98-32cf50e17663), [Sponge](http://sponge2/bdafde32-91d5-48a1-8f98-32cf50e17663)
status: failed
<details><summary>Test output</summary><br><pre>Traceback (most recent call last):
  File "/workspace/appengine/standard_python3/bundled-services/mail/wsgi/main_test.py", line 99, in test_send_receive
    assert expected in text_payloads
AssertionError: assert 'Received greeting for valid-user@e8ee22981d3748e99e0aa9c2da93faeb-dot-cloud-python-ci-resources.appspotmail.com' in ''</pre></details>

### PR DESCRIPTION
**Pull request by engelke**
## Description

Fixes #8212, #8213, #8214

### CODE REVIEW

File: appengine/standard_python3/bundled-services/blobstore/django/main_test.py [hunk 1]
```diff
@@ -15,40 +15,62 @@
...
+
+    Example:
+        result = gcloud_cli("app deploy --no-promote")
+        print(f"Deployed version {result['versions'][0]['id']}")
+
+    Raises Exception with the stderr output of the last attempt if all
+    retries fail.
+    """
+    for retry in range(retries):
+        time.sleep(retry * 5)  # Wait 0, 5, 10, 15, etc seconds before next try
```
Comment (ace-n) [inline appengine/standard_python3/bundled-services/blobstore/django/main_test.py:44]: Nit: should this be "exponential with noise"? i.e.
```
time.sleep(retry ** 2 + randint(1, 10))
```

**Review by ace-n — COMMENTED** [review]
I reviewed the first file; I assume the changes are copy-pasted into the others?

(Either way, I'll take a closer look tomorrow.)

**Review by leahecole — DISMISSED** [review]
Unless there is there a reason we have to use `time` please use `backoff`, which is our preferred library for retries. [This example](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#retry-rpcs) in the authoring guide is based on time, but you can also pass [`max_tries`](https://pypi.org/project/backoff/) instead of `max_time` to do something closer to what I think you're trying to implement

**Review by ace-n — COMMENTED** [review]
There seems to be quite a bit of code duplication here. 🙁 

@engelke is there any chance we can move the duplicative code into some "shared" file, and either
a) symlink to it, _or_
b) import it using Python's path-ing mechanisms?

**Comment by engelke** [comment]
@ace-n I've always had each sample self-contained. I prefer it the way it currently is.

File: appengine/standard_python3/bundled-services/blobstore/django/main_test.py [hunk 2]
```diff
@@ -17,38 +17,55 @@
...
+
+
+@pytest.fixture
+def version():
+    """Launch a new version of the app for testing, and yield the
+    project and version number so tests can invoke it, then delete it.
+    """
+
+    result = gcloud_cli(f"app deploy --no-promote --version={uuid.uuid4().hex}")
+    version_id = result["versions"][0]["id"]
```
Comment (ace-n) [inline appengine/standard_python3/bundled-services/blobstore/django/main_test.py:63]: If `result` is defined but `result["versions"]` is not, will this print out a meaningful error message? (Ideally, one that includes `result` itself?)

**Review by ace-n — APPROVED** [review]
First file LGTM, other than concerns about `result` being logged on failure.

(I'm assuming the other files' changes are similar; @engelke LMK if you'd like me to review them one-by-one. 👍 )

**Pull request merged**